app/javascript/article-form/components/TagsField.jsx

```diff
@@ -1,34 +1,90 @@
 import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
 import PropTypes from 'prop-types';
-import { Tags } from '../../shared/components/tags';
+import { TagAutocompleteOption } from './TagAutocompleteOption';
+import { TagAutocompleteSelection } from './TagAutocompleteSelection';
+import { MultiSelectAutocomplete } from '@crayons';
+import { fetchSearch } from '@utilities/search';
 
-export const DEFAULT_TAG_FORMAT = '[0-9A-Za-z, ]+';
+/**
+ * TagsField for the article form. Allows users to search and select up to 4 tags.
+ *
+ * @param {Function} onInput Callback to sync selections to article form state
+ * @param {string} defaultValue Comma separated list of any currently selected tags
+ * @param {Function} switchHelpContext Callback to switch the help context when the field is focused
+ */
+export const TagsField = ({ onInput, defaultValue, switchHelpContext }) => {
+  const [defaultSelections, setDefaultSelections] = useState([]);
+  const [defaultsLoaded, setDefaultsLoaded] = useState(false);
+  const [topTags, setTopTags] = useState([]);
+
+  useEffect(() => {
+    window
+      .fetch('/tags/suggest')
```

Comment:
Refactor (blocking): `window.` is not necessary. Use `fetch`.

```suggestion
    fetch('/tags/suggest')
```